aspnetcore/fundamentals/configuration.md

```diff
@@ -5,162 +5,981 @@ ms.author: riande
 
-## Using the built-in sources
+It's typical to have different configuration settings for different environments, for example, development, test and production. The following highlighted code hooks up two configuration providers to three sources:
 
-The configuration framework has built-in support for JSON, XML, and INI configuration files, as well as support for in-memory configuration (directly setting values in code) and the ability to pull configuration from environment variables and command line parameters. Developers are not limited to using a single configuration source. In fact several may be set up together such that a default configuration is overridden by settings from another source if they are present.
+1. JSON provider, reading *appsettings.json*
+2. JSON provider, reading *appsettings.<EnvironmentName>.json*
+3. Environment variables provider
 
-Adding support for additional configuration sources is accomplished through extension methods. These methods can be called on a `ConfigurationBuilder` instance in a standalone fashion, or chained together as a fluent API. Both of these approaches are demonstrated in the sample below.
+<!--
+TODO add link
+http://docs.asp.net/projects/api/en/latest/autoapi/Microsoft/Extensions/Configuration/JsonConfigurationExtensions/index.html#Microsoft.Extensions.Configuration.JsonConfigurationExtensions.AddJsonFile)
+-->
+See ``AddJsonFile`` for an explanation of the parameters. Configuration sources are read in the order they are specified. In the code above, the environment variables are read last, any configuration values set through the environment would replace those set in the two previous providers.
 
-<a name=custom-config></a>
+The environment is typically set to one of `Development`, `Staging`, or `Production`. See [Working with Multiple Environments](environments.md) for more information.
 
-[!code-csharp[Main](configuration/sample/src/CustomConfigurationProvider/Program.cs?range=12-25)]
+Configuration considerations:
 
-The order in which configuration sources are specified is important, as this establishes the precedence with which settings will be applied if they exist in multiple locations. In the example below, if the same setting exists in both *appsettings.json* and in an environment variable, the setting from the environment variable will be the one that is used. The last configuration source specified "wins" if a setting exists in more than one location. The ASP.NET team recommends specifying environment variables last, so that the local environment can override anything set in deployed configuration files.
+* The built in configuration providers are not refreshed when the configuration data changes. If your configuration data changes, you'll need to restart your app to get the new data.
```

Comment:
Since the built-in file-based providers do support reloading configuration on changes as an option. Saying otherwise is confusing.